File: python/ProgramStudi.py
from Course import Course

class ProgramStudi ():
  __kode = ''# atribut KODE
  __namaProdi =""# atribut namaProdi
  __course = []# atribut fakultas


  def __init__(self, kode = '', namaProdi ="", course = Course()):
    # inisiasi dengan value awal
    self.__kode = kode
    self.__namaProdi = namaProdi
    self.__course.append(course)
  

  def getKode(self):
    # return KODE mahasiswa
    return self.__kode
  
  def getNamaProdi(self):
    # return namaProdi
    return self.__namaProdi
  
  def getCourse(self):
    #return course
    return self.__course
  
  def setKode(self, kode):
    # menggati value KODE
    self.__kode = kode
  
  def setNamaProdi(self,namaProdi):
    # mengganti value namaProdi
    self.__namaProdi = namaProdi
    
  def addCourse(self, course):
    self.__course.append(course)
  

File: c++/ProgramStudi.cpp
class ProgramStudi {
private:
  string kode; // atribut KODE
  string namaProdi; // atribut namaProdi
  vector<Course> course; // atribut course

public:
  ProgramStudi(string kode = "", string namaProdi = "", Course course = Course()) {
    // inisiasi dengan value awal
    this->kode = kode;
    this->namaProdi = namaProdi;
    this->course.push_back(course);
  }
  string getKode() {
    // return KODE mahasiswa
    return this->kode;
  }

  string getNamaProdi() {
    // return namaProdi
    return this->namaProdi;
  }

  vector<Course> getCourse() {
    // return course
    return this->course;
  }

  void setKode(string kode) {
    // menggati value KODE
    this->kode = kode;
  }

  void setNamaProdi(string namaProdi) {
    // mengganti value namaProdi
    this->namaProdi = namaProdi;
  }

  void addCourse(Course course) {
    this->course.push_back(course);
  }
};